### ISSUE TITLE
Deadlock creating code_systems in Derby

### ISSUE THREAD

**Issue by punktilious** [IBM/FHIR#1417]
**Describe the bug**

Creating new code-systems in parallel can cause deadlocks in Derby. 
```
Caused by: ERROR 40001: A lock could not be obtained due to a deadlock, cycle of locks and waiters is:
Lock : ROW, CODE_SYSTEMS, (1,7)
  Waiting XID : {24547, S} , APP, SELECT code_system_id FROM code_systems WHERE code_system_name = ?
  Granted XID : {24549, X}
Lock : ROW, CODE_SYSTEMS, (1,8)
  Waiting XID : {24549, S} , APP, SELECT code_system_id FROM code_systems WHERE code_system_name = ?
  Granted XID : {24547, X}
```

**To Reproduce**
Create new resources in parallel where those resources require the creation of new code-systems.

**Expected behavior**
All resources loaded successfully without any failures due to deadlocks.

**Additional context**
This appears to be due to its less sophisticated concurrency control and the row-locking behavior of Derby. Db2 and PostgresSQL use more advanced locking mechanisms to avoid this sort of issue.

Derby is not recommended for production use. Because this issue does not occur on Db2 and PostgreSQL, fixing it is lower priority.
